### PR TITLE
added iso to financial period conversion with test

### PR DESCRIPTION
**Pull request by DonShelly**
[ticket reference](https://trello.com/c/he2QutIT/436-fe-list-of-financial-periods-covered-by-the-data)

We only want to show options in the period dropdown for which we have data

Data flow:
BE endpoint (Get earliest and latest submission dates) -->
FE read endpoint response -->
Convert ISO times to Financial quarter equivalents -->
FE endpoint returns 'padded' list of financial years to frontend view -->
list of years rendered in frontend years dropdown

Steps to implement:
do logic for converting date ranges into financial years
*Endpoint would return financial years for which we hold data e.g.
['2020/2021', ‘2021/2022’, ‘2022/2023’]
Convert to financial periods
use endpoint to render dropdown for financial years

PyTest added

### CODE REVIEW

File: tests/test_date_conversions.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from app.main.download_data import generate_financial_years
+
+
+def test_convert_iso_datetime_ranges_to_financial_year_periods():
+    min_date_str = "2019-05-01"
```
Comment (gidsg) [inline tests/test_date_conversions.py:5]: Can we test a time before April to see if it comes under the previous year?

File: app/main/download_data.py [hunk 2]
```diff
@@ -61,8 +60,34 @@
...
+
+    return financial_years
+
+
+def populate_financial_years():
+    # TODO: get dates from BE
+    # hardcoded values:
+    min_date_str = "2019-05-01"
+    max_date_str = "2023-06-01"
+
```
Comment (gidsg) [inline app/main/download_data.py:83]: Can we have it take datetime objects as input arguments instead?

**Comment by cyrusdobbs** [comment]
pre-commit.ci autofix

**Review by gidsg — APPROVED** [review]

**Comment by DonShelly** [comment]
pre-commit.ci autofix

**Comment by DonShelly** [comment]
pre-commit.ci autofix

**Pull request merged**